common/src/stack/command/stack/commands/dump/plugin_host.py

```diff
@@ -0,0 +1,157 @@
+# @copyright@
+# Copyright (c) 2006 - 2018 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+import json
+
+class Plugin(stack.commands.Plugin):
+
+	def provides(self):
+		return 'host'
+
+
+	def run(self, args):
+
+		if args and 'host' not in args:
+			return
+
+		document_prep = {'host':[]}
+
+		#json.loads(Nonetype) fails, so first check that our 'stack list' command returned something.
+		#if not, use an empty list as a placeholder.
+		host_data = self.owner.command('list.host', [ 'output-format=json' ])
+		if host_data:
+			host_data = json.loads(host_data)
+
+			for host in host_data:
+				hostname = host['host']
+				interface_data = self.owner.command('list.host.interface', [ f'{hostname}', 'output-format=json' ])
+				if interface_data:
+					interface_data = json.loads(interface_data)
+				else:
+					interface_data = []
+				interface_alias_data = self.owner.command('list.host.alias', [ f'{hostname}', 'output-format=json' ])
+				if interface_alias_data:
+					interface_alias_data = json.loads(interface_alias_data)
+				else:
+					interface_alias_data = []
+
+				interface_prep = []
+				for interface in interface_data:
+					if interface['host'] == hostname:
+						interface_prep.append({
+								'interface':interface['interface'],
+								'default':interface['default'],
+								'mac':interface['mac'],
+								'ip':interface['ip'],
+								'network':interface['network'],
+								'name':interface['name'],
+								'module':interface['module'],
+								'vlan':interface['vlan'],
+								'options':interface['options'],
+								'channel':interface['channel'],
+								'alias':interface_alias_data
+						})
+				attr_data = self.owner.command('list.host.attr', [ f'{hostname}', 'output-format=json' ])
```

Comment:
`list.host.attr` is relatively expensive.

You should move this lookup to just before the `for host in host_data` loop, do it against all hosts instead of 1 at a time, and then build a dictionary of {hostname: [("attr": "someattr", "value": "someval", etc), (row2...)], hostname2: etc}

Then on line 64 or so you can do a dictionary index instead of looping through until you find the right host.  Build this structure once, then your lookups are fast and your code becomes shorter.

something like:

```
grouped_hosts = {}
for line in lst_host_attr:
  if line['host'] in grouped_hosts:
    grouped_hosts[line['host']].append(line)
  else:
    grouped_hosts[line['host']] = [line]

# now have a dict of hostnames, whose keys are a list of `list.host.attr` rows.
# print(grouped_hosts['infiniband-3-13'][0])
{'host': 'infiniband-3-13', 'scope': 'global', 'type': 'var', 'attr': 'Info_CertificateCountry', 'value': 'US'}

.....

for host in host_data:
  # random other stuff about ifaces and aliases
  attrs = attr_data[hostname]
  for attr in attrs:
    if attr['attr'] == 'metadata':
      # do_stuff
    #etc, etc

```